packages/discord.js-utilities/src/lib/MessagePrompter.ts

```diff
@@ -0,0 +1,485 @@
+		if (!this.reactions?.length) throw new TypeError('There are no reactions provided.');
+
+		const response = await this.collectReactions(channel, authorOrFilter, this.reactions);
+
+		return this.explicitReturn ? response : response.reaction ?? response;
+	}
+}
+
+export class MessagePrompterMessageStrategy extends MessagePrompterBaseStrategy {
+	/**
+	 * The emojis used
+	 */
+	public reactions: string[] | EmojiResolvable[];
+
+	/**
+	 * Constructor for the [[MessagePrompterBaseStrategy]] class
+	 * @param messagePrompter The used instance of [[MessagePrompter]]
+	 * @param options Overrideable options if needed.
+	 */
+	public constructor(message: MessagePrompterMessage, options: IMessagePrompterReactionStrategyOptions) {
+		super('message', options, message);
+
+		this.reactions = options?.reactions;
+	}
+
+	/**
+	 * This executes the [[MessagePrompter]] and sends the message if [[IMessagePrompterOptions.type]] equals message.
+	 * The handler will wait for one (1) message.
+	 * @param channel The channel to use.
+	 * @param authorOrFilter An author object to validate or a {@link https://discord.js.org/#/docs/main/stable/typedef/CollectorFilter CollectorFilter} predicate callback.
+	 * @returns A promise that resolves to the message object received.
+	 */
+	public async run(
+		channel: TextChannel | NewsChannel | DMChannel,
+		authorOrFilter: User | CollectorFilter
+	): Promise<IMessagePrompterExplicitMessageReturn | Message> {
+		this.appliedMessage = await channel.send(this.message);
+
+		const collector = await channel.awaitMessages(this.createMessagePromptFilter(authorOrFilter), {
+			max: 1,
+			time: this.timeout,
+			errors: ['time']
+		});
+		const response = collector.first();
+
+		if (!response) {
+			throw new Error('No messages received');
+		}
+
+		return this.explicitReturn
+			? {
+					response,
+					strategy: this as MessagePrompterBaseStrategy,
+					appliedMessage: this.appliedMessage,
+					message: this.message
+			  }
+			: response;
+	}
+
+	/**
+	 * Creates a filter for the collector to filter on
+	 * @return The filter for awaitMessages function
+	 */
+	private createMessagePromptFilter(authorOrFilter: User | CollectorFilter): CollectorFilter {
+		return async (message: Message) =>
+			(typeof authorOrFilter === 'function' ? await authorOrFilter(message) : message.author.id === authorOrFilter.id) && !message.author.bot;
+	}
+}
+
+/**
+ * This is a [[MessagePrompter]], a utility that sends a prompt message that can resolve any kind of input.
+ * There are several specifiable types to prompt for user input, and they are as follows:
+ * - Confirm
+ *   This will send a simple Yes/No prompt, using reactions.
+ * - Number
+ *   This will prompt for an integer. You can specify a range, however by default it will use 0 - 10.
+ * - Reactions
+ *   This can be any kind of reaction emoji Discord supports and as many as you want. This type will return that reaction instead of a boolean.
```

Comment:
```suggestion
 *   This can be any kind of reaction emoji that Discord supports, and as many as you want. This type will return that reaction instead of a boolean.
```